### ISSUE TITLE
NUnit1001 false positive for types with custom type converters

### ISSUE THREAD

**Issue by jnm2** [nunit/nunit.analyzers#232]
A warning should not be produced. The analyzer can see that CustomType has `[TypeConverter(typeof(CustomTypeConverter))]` even though it can't run it. It's a safe assumption that type converters can convert from `string`. Even without that assumption (say, with an `int` test parameter), the warning message is making a claim that it can't prove.

```cs
using System;
using System.ComponentModel;
using System.Globalization;
using NUnit.Framework;

class C
{
    // NUnit1001 The value of the argument at position '0' of type string cannot be assigned to
    // the argument 'customType' of type CustomType.
    //        ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
    [TestCase("String representation")]
    public void M(CustomType customType)
    {
    }
}

[TypeConverter(typeof(CustomTypeConverter))]
struct CustomType
{
    public CustomType(string value)
    {
        Value = value;
    }

    public string Value { get; }

    private sealed class CustomTypeConverter : TypeConverter
    {
        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
        {
            return sourceType == typeof(string) || base.CanConvertFrom(context, sourceType);
        }

        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
        {
            return value is string str
                ? new CustomType(str)
                : base.ConvertFrom(context, culture, value);
        }

        public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
        {
            return destinationType == typeof(string) || base.CanConvertTo(context, destinationType);
        }

        public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
        {
            return destinationType == typeof(string)
                ? value.ToString()
                : base.ConvertTo(context, culture, value, destinationType);
        }
    }
}
```

**Comment by mikkelbu** [comment]
@jnm2 This sounds like a duplicate of #197, but I like this issue better as it provides more information.

**Comment by jnm2** [comment]
Oh dear, thanks. Can I close this one and move my comment there?

**Comment by mikkelbu** [comment]
Sure 👍
